src/util_types/mutator_set/ms_membership_proof.rs

```diff
@@ -201,38 +232,45 @@ impl MsMembershipProof {
             &mut mmr::mmr_membership_proof::MmrMembershipProof<Hash>,
         > = vec![];
 
-        // The `mmr_membership_proof_index_to_membership_proof_index` variable is to remember
-        // which parts of the MMR membership proofs that map to MS membership proofs. This is
+        // The `mmr_mp_index_to_ms_mp_index` variable remembers
+        // which MMR membership proofs that map to MS membership proofs. This is
         // required to return the indices of the MS membership proofs that have been updated
         // by this function call.
-        let mut mmr_membership_proof_index_to_membership_proof_index: Vec<usize> = vec![];
+        let mut mmr_mp_index_to_ms_mp_index = vec![];
+        let mut mmr_membership_indices = vec![];
         for (i, mp) in membership_proofs.iter_mut().enumerate() {
             if mps_for_batch_append.contains(&i) {
-                for (_, (mmr_mp, _chnk)) in mp.target_chunks.iter_mut() {
-                    mmr_membership_proofs_for_append.push(mmr_mp);
-                    mmr_membership_proof_index_to_membership_proof_index.push(i);
+                for (chunk_index, (mmr_mp, _chunk)) in mp.target_chunks.iter_mut() {
+                    if *chunk_index != old_window_start_batch_index {
```

Comment:
This condition seems new. I think it's covered by line 203: `if chunk_index < old_window_start_batch_index`

Reply:
I think it's not covered because if I drop this condition then tests fail. That said, it certainly looks as though it should be covered by this line. So I am left wondering why it's not and what we are overlooking and whether a soundness or completeness error is lurking there.